### ISSUE TITLE
E: DT_JSON's custom field, set, should be return the raw data first

### ISSUE THREAD

**Issue by chez14** [ikkez/f3-cortex#49]
The `DT_JSON` type field, the `set_*` one should return the raw data first.

Like an array, we want to handle the array as is, like adding some prefix onto each of them or validating those array first before it saved into the database. (most of case(s))

some example (my code; sorry, it's not perfect)

```php
    function set_jawaban($jawaban){
        $jawaban = array_map(function($data){
            //todo: Rule validation.
            return [
                'id'=> $data['id'],
                'answer'=>$data['answer']
            ];
        }, (array)json_decode($jawaban, true));

        return json_encode($jawaban);
    }
```
The `$jawaban` variable are returning JSON-encoded string, so when I need to edit it back, I have to decode it and return it by encoding it.

That's why i need to `json_decode` and `json_encode` again. It's not efficient  😢.

Thank you

**Comment by ikkez** [comment]
fixed in the upcoming v1.6

**Comment by chez14** [comment]
i even forgot this issue hahah, thanks for the update btw :+1:
